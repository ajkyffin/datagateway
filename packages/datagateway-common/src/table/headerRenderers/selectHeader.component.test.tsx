import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';
import SelectHeader from './selectHeader.component';

describe('Select column header component', () => {
  let shallow;
  const setLastChecked = jest.fn();
  const onCheck = jest.fn();
  const onUncheck = jest.fn();
  const selectHeaderProps = {
    dataKey: 'test',
    className: 'test-class',
    selectedRows: [],
    totalRowCount: 3,
    onCheck,
    onUncheck,
    allIds: [1, 2, 3],
  };

  beforeEach(() => {
    shallow = createShallow({ untilSelector: 'div' });
  });

  afterEach(() => {
    setLastChecked.mockClear();
    onCheck.mockClear();
    onUncheck.mockClear();
  });

  it('renders correctly when unchecked', () => {
    const wrapper = shallow(<SelectHeader {...selectHeaderProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly when indeterminate', () => {
    const wrapper = shallow(
      <SelectHeader {...selectHeaderProps} selectedRows={[1]} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly when checked', () => {
    const wrapper = shallow(
      <SelectHeader {...selectHeaderProps} selectedRows={[1, 2, 3]} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('calls onCheck when not all rows are selected and the checkbox is clicked', () => {
    const wrapper = shallow(
      <SelectHeader {...selectHeaderProps} selectedRows={[1]} />
    );

    wrapper.childAt(0).prop('onClick')();
    expect(onCheck).toHaveBeenCalledWith([1, 2, 3]);
  });

  it('calls onUncheck when all rows are selected and the checkbox is clicked', () => {
    const wrapper = shallow(
      <SelectHeader {...selectHeaderProps} selectedRows={[1, 2, 3]} />
    );

    wrapper.childAt(0).prop('onClick')();
    expect(onUncheck).toHaveBeenCalledWith([1, 2, 3]);
  });
});
