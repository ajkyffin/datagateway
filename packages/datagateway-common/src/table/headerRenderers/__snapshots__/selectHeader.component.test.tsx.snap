// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select column header component renders correctly when checked 1`] = `
<div
  aria-sort={null}
  className="MuiTableCell-root MuiTableCell-head test-class MuiTableCell-paddingCheckbox MuiTableCell-sizeSmall"
  role="cell"
>
  <WithStyles(ForwardRef(Checkbox))
    checked={true}
    checkedIcon={
      <Memo
        fontSize="small"
      />
    }
    disabled={false}
    icon={
      <Memo
        fontSize="small"
      />
    }
    indeterminate={false}
    indeterminateIcon={
      <Memo
        fontSize="small"
      />
    }
    inputProps={
      Object {
        "aria-label": "select all rows",
      }
    }
    onClick={[Function]}
    size="small"
    style={
      Object {
        "height": 20,
        "margin": "auto",
      }
    }
  />
</div>
`;

exports[`Select column header component renders correctly when indeterminate 1`] = `
<div
  aria-sort={null}
  className="MuiTableCell-root MuiTableCell-head test-class MuiTableCell-paddingCheckbox MuiTableCell-sizeSmall"
  role="cell"
>
  <WithStyles(ForwardRef(Checkbox))
    checked={false}
    checkedIcon={
      <Memo
        fontSize="small"
      />
    }
    disabled={false}
    icon={
      <Memo
        fontSize="small"
      />
    }
    indeterminate={true}
    indeterminateIcon={
      <Memo
        fontSize="small"
      />
    }
    inputProps={
      Object {
        "aria-label": "select all rows",
      }
    }
    onClick={[Function]}
    size="small"
    style={
      Object {
        "height": 20,
        "margin": "auto",
      }
    }
  />
</div>
`;

exports[`Select column header component renders correctly when loading is true 1`] = `
<div
  aria-sort={null}
  className="MuiTableCell-root MuiTableCell-head test-class MuiTableCell-paddingCheckbox MuiTableCell-sizeSmall"
  role="cell"
>
  <WithStyles(ForwardRef(Checkbox))
    checked={false}
    checkedIcon={
      <Memo
        fontSize="small"
      />
    }
    disabled={true}
    icon={
      <Memo
        fontSize="small"
      />
    }
    indeterminateIcon={
      <Memo
        fontSize="small"
      />
    }
    inputProps={
      Object {
        "aria-label": "select all rows",
      }
    }
    onClick={[Function]}
    size="small"
    style={
      Object {
        "height": 20,
        "margin": "auto",
      }
    }
  />
</div>
`;

exports[`Select column header component renders correctly when selectedRows is undefined 1`] = `
<div
  aria-sort={null}
  className="MuiTableCell-root MuiTableCell-head test-class MuiTableCell-paddingCheckbox MuiTableCell-sizeSmall"
  role="cell"
>
  <WithStyles(ForwardRef(Checkbox))
    checked={false}
    checkedIcon={
      <Memo
        fontSize="small"
      />
    }
    disabled={true}
    icon={
      <Memo
        fontSize="small"
      />
    }
    indeterminateIcon={
      <Memo
        fontSize="small"
      />
    }
    inputProps={
      Object {
        "aria-label": "select all rows",
      }
    }
    onClick={[Function]}
    size="small"
    style={
      Object {
        "height": 20,
        "margin": "auto",
      }
    }
  />
</div>
`;

exports[`Select column header component renders correctly when unchecked 1`] = `
<div
  aria-sort={null}
  className="MuiTableCell-root MuiTableCell-head test-class MuiTableCell-paddingCheckbox MuiTableCell-sizeSmall"
  role="cell"
>
  <WithStyles(ForwardRef(Checkbox))
    checked={false}
    checkedIcon={
      <Memo
        fontSize="small"
      />
    }
    disabled={false}
    icon={
      <Memo
        fontSize="small"
      />
    }
    indeterminate={false}
    indeterminateIcon={
      <Memo
        fontSize="small"
      />
    }
    inputProps={
      Object {
        "aria-label": "select all rows",
      }
    }
    onClick={[Function]}
    size="small"
    style={
      Object {
        "height": 20,
        "margin": "auto",
      }
    }
  />
</div>
`;
